example/cajita_tutorial/09_grid_parallel/grid_parallel_example.cpp

```diff
@@ -0,0 +1,207 @@
+    std::array<double, 3> global_high_corner = {
+        global_low_corner[0] + cell_size * global_num_cell[0],
+        global_low_corner[1] + cell_size * global_num_cell[1],
+        global_low_corner[2] + cell_size * global_num_cell[2] };
+    auto global_mesh = Cajita::createUniformGlobalMesh(
+        global_low_corner, global_high_corner, global_num_cell );
+
+    // Create the global grid.
+    auto global_grid = Cajita::createGlobalGrid( MPI_COMM_WORLD, global_mesh,
+                                                 is_dim_periodic, partitioner );
+
+    // Create an array layout on the cells.
+    int halo_width = 2;
+    int dofs_per_cell = 4;
+    auto cell_layout = Cajita::createArrayLayout(
+        global_grid, halo_width, dofs_per_cell, Cajita::Cell() );
+    auto local_grid = cell_layout->localGrid();
+
+    // Create an array.
+    std::string label( "example_array" );
+    auto array = Cajita::createArray<double, device_type>( label, cell_layout );
+    //-----------------------------------------------------------------------//
+
+    /*
+      The grid parallel functions first take a kernel label, then an execution
+      space rather than a direct Kokkos range policy. Internally, a
+      multidimensional range policy (Kokkos::MDRangePolicy) is created with the
+      execution space given. Next, the Own or Ghost tag is passed to define
+      whether the kernel should include ghosted entities or not (remember that
+      specifying Ghost includes both owned and ghosted entities). The mesh
+      entity is passed next and finally, the parallel kernel.
+
+      Here we use a lambda function for simplicity, but a functor with or
+      without tag arguments are also options. Note that the kernel signature
+      uses the three indicies of the 3D grid. We set every value on the cells in
+      the grid to 1.
+    */
+    auto array_view = array->view();
+    Cajita::grid_parallel_for(
+        "local_grid_for", exec_space(), *local_grid, Cajita::Ghost(),
+        Cajita::Cell(), KOKKOS_LAMBDA( const int i, const int j, const int k ) {
+            for ( int l = 0; l < 4; ++l )
+                array_view( i, j, k, l ) = 1.0;
+        } );
+
+    /*
+      We can similarly do grid reductions with the same arguments as
+      parallel_for; the only change is the reduction variable in the kernel
+      signature and the final return value.
+    */
+    double sum = 0.0;
+    Cajita::grid_parallel_reduce(
+        "local_grid_reduce", exec_space(), *local_grid, Cajita::Ghost(),
+        Cajita::Cell(),
+        KOKKOS_LAMBDA( const int i, const int j, const int k, double& result ) {
+            for ( int l = 0; l < 4; ++l )
+                result += array_view( i, j, k, l );
+        },
+        sum );
+
+    /*
+      Here, the total should match the total grid size (including ghosts) times
+      the four field values for each grid point.
+    */
+    auto ghost_is = local_grid->indexSpace( Cajita::Ghost(), Cajita::Cell(),
```

Comment:
Show a version that uses a 4D index space as well that you get from the `ArrayLayout`